### PR TITLE
Exclusion Hunter V: DefaultCharset

### PR DESCRIPTION
**Pull request by Jolyon-S**
**Goals (and why)**:
Continue the hunt.

**Implementation Description (bullets)**:
Use `StandardCharsets.UTF_8`.

**Where should we start reviewing?**:
diff

**Priority (whenever / two weeks / yesterday)**:
whenever

### CODE REVIEW

File: lock-impl/src/main/java/com/palantir/lock/logger/LockStateYamlWriter.java [hunk 1]
```diff
@@ -43,7 +44,8 @@
      * Creates a LockStateYamlWriter for the given file in append mode.
      */
     public static LockStateYamlWriter create(File file) throws IOException {
-        return new LockStateYamlWriter(new BufferedWriter(new FileWriter(file, true)));
+        return new LockStateYamlWriter(Files.newBufferedWriter(
+                file.toPath(), StandardCharsets.UTF_8, StandardOpenOption.CREATE, StandardOpenOption.APPEND));
```
Comment (Jolyon-S) [inline lock-impl/src/main/java/com/palantir/lock/logger/LockStateYamlWriter.java:48]: This is probably the only change we want to sanity check

Comment (sudiksha27) [reply lock-impl/src/main/java/com/palantir/lock/logger/LockStateYamlWriter.java:48]: Is this file ever read? Can you write a test and see if this works locally?

Comment (Jolyon-S) [reply lock-impl/src/main/java/com/palantir/lock/logger/LockStateYamlWriter.java:48]: I have literally no idea. Could do

Comment (Jolyon-S) [reply lock-impl/src/main/java/com/palantir/lock/logger/LockStateYamlWriter.java:48]: I believe `LockServiceTest` tests this code path

**Review by sudiksha27 — APPROVED** [review]

**Pull request merged**